frontend/src/components/RobotCards/RobotStatusView.tsx

```diff
@@ -33,14 +33,21 @@ export function RobotStatusSection() {
     }, [])
 
     const updateRobots = () => {
-        apiCaller.getRobots().then((result) => {
-            setRobots(result)
+        apiCaller.getRobots().then((result: Robot[]) => {
+            setRobots(sortRobotsByStatus(result))
         })
     }
 
     var robotDisplay = robots.map(function (robot) {
         return <RobotStatusCard key={robot.id} robot={robot} />
     })
+    const sortRobotsByStatus = (robots: Robot[]): Robot[] => {
+        const sortedRobots = robots.sort((robot, robotToCompareWith) =>
+            robot.status! > robotToCompareWith.status! ? 1 : -1
```

Comment:
Since robot.status is required in the backed data model , this assumption is OK in my mind.